test/pynrn/test_mpi_init.py

```diff
@@ -0,0 +1,30 @@
+from neuron import h
+import sys
+import subprocess
+
+
+def srun(cmd):
+    print("--------------------")
+    print(cmd)
+    subprocess.run(cmd, shell=True).check_returncode()
+
+
+# test various ways to launch nrniv and python that can MPI_INIT
+def test_mpi_init():
+    python = sys.executable + " "
+    nrniv = "nrniv "
+    nompi = "NRN_ENABLE_MPI=OFF" in h.nrnversion(6)
+    mpiexec = "mpiexec -n 1 "
+    env = "env NEURON_INIT_MPI=1 "
+    if not nompi:
+        srun(mpiexec + nrniv + "-mpi -c 'quit()'")
+        srun(mpiexec + python + "-c 'from neuron import h; h.nrnmpi_init(); h.quit()'")
+        srun(env + mpiexec + python + "-c 'from neuron import h; h.quit()'")
+    srun(nrniv + "-mpi -c 'quit()'")
+    srun(nrniv + "-c 'nrnmpi_init()' -c 'quit()'")
+    srun(python + "-c 'from neuron import h; h.nrnmpi_init(); h.quit()'")
+    srun(env + python + "-c 'from neuron import h; h.quit()'")
```

Comment:
Should we do this via CMake rather than launching via python subprocess? (because hardcoding mpiexec and other params in the code is not portable in all scenarios)

Reply:
Good question!  Environment dependent tests seem presently very brittle. For example if one does ```make``` without a ```make install```, then ```make test``` could get the wrong ```nrniv``` above because ```/.../build/bin``` is not added to the PATH.  (with ```make install```, then ```/.../build/install/bin``` IS added to the path.) Anyway, it seems that test scripts occasionally need to to know various details about the cmake environment. One thing about python scripts is that it's generally possible to figure out those details. (or fix up cmake so the the test system provides what is needed, e,g. proper PATH).  For this PR, the segfault was never seen except on linux with openmpi. So perhaps the test needs restriction to that. The error on mac's may be acceptable since the  error message is explicit about what is wrong.

Reply:
Yes. This PR needs significant changes to work correctly.

Reply:
I **think** 136caced1 fixes the underlying problem in [inithoc.cpp](https://github.com/neuronsimulator/nrn/compare/4a66a2004a7b0833cd30168ce4a2a0d472c84a08...136caced1358751d23c2f33fce446a617ec952cf#diff-f5efc9cdc53554965222bc4803e239ca4516c8d4e4c59df12943e3100a4675a0). I will look at the tests tomorrow.